config/v1/types_apiserver.go

```diff
@@ -0,0 +1,24 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// APIServer holds cluster-wide information about api-servers.  The canonical name is `cluster`
+type APIServer struct {
```

Comment:
@openshift/api-reviewers  let's argue over names and what is included.

Reply:
Is this for the `kube-apiserver`, `openshift-apiserver`, or both?

Reply:
GenericAPIServerConfig

Reply:
* There is already an [existing `GenericAPIServerConfig` type](https://github.com/openshift/api/blob/d22df4f5cec775b9d689711da4d193766aa50dfe/config/v1/types.go#L222).
* None of the other cluster config types end in `Config`
* Some more suggestions:
  * `APIServerTLS` (if TLS certs are all that end up in here)
  * `APIServerCommon`
  * `APIServers`

Reply:
:+1: for the current name, it's short and clean.

Reply:
> Is this for the `kube-apiserver`, `openshift-apiserver`, or both?

KubeAPIServer, but I'm ok claiming the name `APIServer` here.



> * `APIServerTLS` (if TLS certs are all that end up in here)
> * `APIServerCommon`
> * `APIServers`

tls is too specific, common is just another way of saying stuff, plural implies a thing we won't do